File: backend/routers/messages.py
"""
消息管理路由
提供单条消息的删除功能。
"""
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy import select
from sqlalchemy.ext.asyncio import AsyncSession

from database import get_db
from models import Message

router = APIRouter(prefix="/api/messages", tags=["消息管理"])


@router.delete("/{message_id}", status_code=204)
async def delete_message(message_id: str, db: AsyncSession = Depends(get_db)):
    """删除单条消息"""
    stmt = select(Message).where(Message.id == message_id)
    result = await db.execute(stmt)
    message = result.scalar_one_or_none()
    
    if not message:
        raise HTTPException(status_code=404, detail="消息不存在")
    
    await db.delete(message)

File: server/routes/messages.js
/**
 * 消息管理路由
 * 提供单条消息的删除功能。
 * 
 * 替代 Python 版 routers/messages.py
 */
const { Router } = require('express');
const { getDb } = require('../database');

const router = Router();

/**
 * DELETE /api/messages/:id - 删除单条消息
 */
router.delete('/:id', (req, res) => {
    const db = getDb();
    const message = db.prepare('SELECT * FROM messages WHERE id = ?').get(req.params.id);

    if (!message) {
        return res.status(404).json({ detail: '消息不存在' });
    }

    db.prepare('DELETE FROM messages WHERE id = ?').run(req.params.id);
    res.status(204).end();
});

module.exports = router;
